ledger/accountdb.go

```diff
@@ -332,43 +354,48 @@ func accountsDbInit(r db.Queryable, w db.Queryable) (*accountsDbQueries, error)
 	return qs, nil
 }
 
-func (qs *accountsDbQueries) listAssets(maxAssetIdx basics.AssetIndex, maxResults uint64) (results []basics.CreatableLocator, err error) {
+func (qs *accountsDbQueries) listCreatables(maxIdx basics.CreatableIndex, maxResults uint64, ctype basics.CreatableType) (results []basics.CreatableLocator, err error) {
 	err = db.Retry(func() error {
 		// Query for assets in range
-		rows, err := qs.listAssetsStmt.Query(maxAssetIdx, maxResults)
+		rows, err := qs.listCreatablesStmt.Query(maxIdx, ctype, maxResults)
 		if err != nil {
 			return err
 		}
 		defer rows.Close()
 
 		// For each row, copy into a new CreatableLocator and append to results
 		var buf []byte
-		var al basics.CreatableLocator
+		var cl basics.CreatableLocator
 		for rows.Next() {
-			err := rows.Scan(&al.Index, &buf)
+			err := rows.Scan(&cl.Index, &buf)
 			if err != nil {
 				return err
 			}
-			copy(al.Creator[:], buf)
-			results = append(results, al)
+			copy(cl.Creator[:], buf)
+			cl.Type = ctype
+			results = append(results, cl)
 		}
 		return nil
 	})
 	return
 }
 
-func (qs *accountsDbQueries) lookupAssetCreator(assetIdx basics.AssetIndex) (addr basics.Address, err error) {
+func (qs *accountsDbQueries) lookupCreator(cidx basics.CreatableIndex, ctype basics.CreatableType) (addr basics.Address, ok bool, err error) {
 	err = db.Retry(func() error {
 		var buf []byte
-		err := qs.lookupAssetCreatorStmt.QueryRow(assetIdx).Scan(&buf)
+		err := qs.lookupCreatorStmt.QueryRow(cidx, ctype).Scan(&buf)
 
+		// Common error: creatable does not exist
 		if err == sql.ErrNoRows {
-			err = fmt.Errorf("asset %d does not exist or has been deleted", assetIdx)
+			return nil
 		}
 
+		// Some other database error
 		if err != nil {
 			return err
 		}
+
+		ok = true
```

Comment:
Consider adding an `CreatableDoesNotExistError` and returning that in `err` instead of adding an `ok bool`.